src/gc.rs

```diff
@@ -39,41 +39,99 @@ use crate::GC_ALLOCATOR;
 /// free `Gc` values behind the scenes.
 #[derive(PartialEq, Eq, Debug)]
 pub struct Gc<T: ?Sized> {
-    pub(crate) objptr: NonNull<GcBox<T>>,
+    pub(crate) ptr: NonNull<GcBox<T>>,
+    _phantom: PhantomData<T>,
 }
 
 impl<T> Gc<T> {
     /// Constructs a new `Gc<T>`.
     pub fn new(v: T) -> Self {
         Gc {
-            objptr: unsafe { NonNull::new_unchecked(GcBox::new(v)) },
+            ptr: unsafe { NonNull::new_unchecked(GcBox::new(v)) },
+            _phantom: PhantomData,
         }
     }
+
+    /// Constructs a new `Gc<MaybeUninit<T>>` which is capable of storing data
+    /// up-to the size permissible by `layout`.
+    ///
+    /// This can be useful if you want to store a value with a custom layout,
+    /// but have the collector treat the value as if it were T.
+    ///
+    /// If `layout` is insufficiently sized to store `T`, a Gc value will not be
```

Comment:
The first sentence here is a bit too formal and hard to parse. Can we just say "`layout` must be at least as large as, and of the same alignment or bigger, as `T`." or similar? I then think we can remove the "this is because" sentence (in a sense it doesn't matter: the user *must* satisfy the contract, and "why" is often more detail than they need).

Reply:
Fixed: 55aee509d1af3eb40a10836370a0123aefc8b1dc